internal/server/boltdbstate/task.go

```diff
@@ -79,6 +79,40 @@ func (s *State) TaskDelete(ref *pb.Ref_Task) error {
 	return err
 }
 
+// TaskCancel deletes a task by reference.
+// NOTE(briancain): this way means each cancel is its own transaction and commit, not the greatest.
+// Previously I attempted to implement this with taskCancel where the entire job
+// triple was canceled inside a single transaction. This ended up deadlocking because
+// when we cancel a job, we also call a `db.Update` on each job when we update its
+// state in the database. This caused a deadlock. For now we cancel each job
+// separately.
+func (s *State) TaskCancel(ref *pb.Ref_Task) error {
+	task, err := s.TaskGet(ref)
+	if err != nil {
+		return err
+	}
+
+	s.log.Debug("canceling start job for task by id", "task id", task.Id, "start job id", task.StartJob.Id)
```

Comment:
Super-nit: "for task by id" seems superfluous for these debug messages.

Reply:
Agreed! I'll push a fix